File: Cocoapods/ObjectiveC/ObjectiveCDemo/ViewController.m
//
//  ViewController.m
//  ObjectiveCDemo
//
//  Created by Maxime Bokobza on 03/03/17.
//  Copyright © 2017 Streamroot. All rights reserved.
//

#import "ViewController.h"
#import <AVFoundation/AVFoundation.h>
#import <StreamrootSDK/StreamrootSDK.h>


@interface ViewController () <StreamrootSDKDelegate>

@property (nonatomic, strong) StreamrootSDK *streamrootSDK;

@end


@implementation ViewController

- (void)viewDidLoad {
    [super viewDidLoad];

    NSString *manifestUrl = @"https://wowza-cloudfront.streamroot.io/liveorigin/stream4/playlist.m3u8";
    self.streamrootSDK = [[StreamrootSDK alloc] initWithStreamrootKey:<#StreamrootKey#> manifestURL:manifestUrl];

    self.streamrootSDK.delegate = self;

    NSError *error;
    BOOL started = [self.streamrootSDK start:&error];
    if (error || !started) {
        // Handle error
        NSLog(@"error: %@", error);
    }

    NSURL *url = [[NSURL alloc] initWithString:[self.streamrootSDK manifestLocalURL]];
    self.player = [[AVPlayer alloc] initWithURL:url];
    [self.view setNeedsLayout];
    [self.view layoutIfNeeded];
    [self.streamrootSDK displayStatsOnView:self.contentOverlayView];
}

- (void)viewDidDisappear:(BOOL)animated {
  [self.streamrootSDK stop];
}

- (double)playbackTime {
    return CMTimeGetSeconds([self.player currentTime]);
}

- (NSArray<NSValue *> *)loadedTimeRanges {
    if (self.player == nil) {
        return [NSArray array];
    }

    NSMutableArray *timeRanges = [NSMutableArray array];
    for (NSValue *value in [[self.player currentItem] loadedTimeRanges]) {
        TimeRange *timeRange = [[TimeRange alloc] initWithRange:[value CMTimeRangeValue]];
        [timeRanges addObject:[[NSValue alloc] initWithTimeRange:timeRange]];
    }
    return timeRanges;
}

- (void)updatePeakBitRate:(double)bitRate {
    
}


@end

File: Carthage/Swift/SwiftDemo/ViewController.swift
//
//  ViewController.swift
//  SwiftDemo
//
//  Created by Maxime Bokobza on 02/03/17.
//  Copyright © 2017 Streamroot. All rights reserved.
//

import AVFoundation
import StreamrootSDK
import AVKit

class ViewController: AVPlayerViewController, StreamrootSDKDelegate {

    internal var streamrootSDK: StreamrootSDK?
    internal let manifestUrl = "https://wowza-cloudfront.streamroot.io/liveorigin/stream4/playlist.m3u8"

    required init?(coder aDecoder: NSCoder) {
        super.init(coder: aDecoder)

        self.streamrootSDK = StreamrootSDK(streamrootKey: <#streamrootKey#>, manifestURL: self.manifestUrl)
        self.streamrootSDK?.delegate = self

        do {
            try self.streamrootSDK?.start()
        } catch let error {
            print("\(error)")
        }
    }

    override func viewDidLoad() {
        super.viewDidLoad()

        if let localPath = self.streamrootSDK?.manifestLocalURL() {
            if let url = URL(string: localPath) {
                self.player = AVPlayer(url: url)
                self.player!.play()
            }
        }
    }

    func playbackTime() -> Double {
        if let player = self.player {
            return CMTimeGetSeconds(player.currentTime())
        }
        return 0
    }

    func loadedTimeRanges() -> [NSValue] {
        guard let player = self.player else {
            return []
        }

        let timeRanges = player.currentItem!.loadedTimeRanges
        return timeRanges.map { (value) -> NSValue in
            NSValue(timeRange: TimeRange(range: value.timeRangeValue))
        }
    }

}
